src/Test/Utilities/Portable/Platform/Custom/MetadataSignatureHelper.cs

```diff
@@ -539,10 +542,14 @@ static private void AppendFieldInfo(FieldInfo field, StringBuilder sb)
 
             if (field.IsInitOnly) sb.Append("initonly ");
             if (field.IsLiteral) sb.Append("literal ");
+#if NET46 || NET461
```

Comment:
I'm a little worried about enumerating frameworks, as we'll likely introduce more and APIs will be added. Would it make sense to turn this into a negative (skip for frameworks we know don't support the API)?

Reply:
I dislike enumerating frameworks as well. The pattern I'm using here is the established pattern though for this entire project. 

May try and hide this via extension methods. We have a couple files that try and encapsulate this madness a bit. 

---
In reply to: [171423595](https://github.com/dotnet/roslyn/pull/25050#discussion_r171423595) [](ancestors = 171423595)

Reply:
Actually I think I can remove this entirely. These APIs are available on NetCoreApp2.0 which is what we are using here. The problem is that our references are simply busted and I need to change the build a bit. It's bit of an invasive change though so I'm goin to do that with a follow up PR. 

[https://github.com/dotnet/roslyn/issues/25146](https://github.com/dotnet/roslyn/issues/25146)

---
In reply to: [171429117](https://github.com/dotnet/roslyn/pull/25050#discussion_r171429117) [](ancestors = 171429117,171423595)